src/types.rs

```diff
@@ -167,6 +167,19 @@ pub struct WeightedUtxo {
     pub utxo: Utxo,
 }
 
+/// A [`Script`] with the `satisfaction_weight` of the resulting TxOut, if it were spend by the
+/// wallet owning the produced UTXO originated from it.
+#[derive(Debug, Clone, PartialEq)]
+pub struct WeightedScript {
+    /// The weight of the witness data and `scriptSig` expressed in [weight units]. This is used to
+    /// properly compute the cost of change for waste metric.
+    ///
+    /// [weight units]: https://en.bitcoin.it/wiki/Weight_units
+    pub satisfaction_weight: usize,
+    /// The script_pubkey
+    pub script_pubkey: Script,
+}
```

Comment:
It was becoming too many comments and mostly just nits, so did a suggestion instead.. How about we have something like below to express this struct? 

Rationales:
 - Cleaner to specify explainer details in the struct doc than in field doc.
 - Its not really a `Script`, its a `script_pubkey`, which is also `Script` type. So all `script_punbkeys` are `Script`, but not all `Script`s are `script_pubkey`. Thus seems better to me as `WeightedScriptPubkey`.  

```suggestion
/// An wallet owned script_pubkey with the weight of the `witness` + `scriptSig` in [weight units] to spend from it.
/// This is useful for coninselection algorithm to to optimize for reducing waste metric.
/// [weight units]: https://en.bitcoin.it/wiki/Weight_units 
#[derive(Debug, Clone, PartialEq)]
pub struct WeightedScriptPubkey {
    /// The weight of the `witness` and `scriptSig` expressed in [weight units].
    pub satisfaction_weight: usize,
    /// The script_pubkey
    pub script_pubkey: Script,
}
```

Reply:
Agree. I'll apply the changes locally to avoid conflicts (I've several changes in the backlog).